### PR TITLE
Add xiaomi credential

### PR DESCRIPTION
**Pull request by weihengSu2**
Things to consider before you submit the PR:

* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description
The PR will add XiaomiCredentials to the Azure NotificationHub SDK.Net repo, which will be used as part of the auth header to send Xiaomi Notifications which are in another PR

### CODE REVIEW

**Comment by jessHuh** [comment]
I left some comments. Could we also add some tests? Thanks!

**Comment by weihengSu2** [comment]
> I left some comments. Could we also add some tests? Thanks!

We don't have unit tests for Credential classes like ApnsCredential or other Credential. But I could try to add to a program.cs or add a new unit test class for Credential?

File: src/Microsoft.Azure.NotificationHubs/NotificationHubDescription.cs [hunk 1]
```diff
@@ -326,6 +326,20 @@ public MpnsCredential MpnsCredential
...
         }
 
+        /// <summary>
+        /// Gets or sets the Xiaomi credential.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The Xiaomi credential.
+        /// </returns>
+        [DataMember(Name = ManagementStrings.XiaomiCredential, IsRequired = false, EmitDefaultValue = false, Order = 1007)]
```
[the remark this answers is not in the document]
Comment (weihengSu2) [reply src/Microsoft.Azure.NotificationHubs/NotificationHubDescription.cs:336]: Thanks! I will update it. Real quick

File: src/Microsoft.Azure.NotificationHubs/XiaomiCredential.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    /// Represents a Xiaomi credential.
+    /// </summary>
+    [DataContract(Name = ManagementStrings.XiaomiCredential, Namespace = ManagementStrings.Namespace)]
+    public class XiaomiCredential : PnsCredential
+    {
+        internal const string AppPlatformName = "xiaomi";
+        internal const string XiaomiEndpointName = "Endpoint";
+        internal const string ProdChineseMainlandEndpoint = @"https://api.xmpush.xiaomi.com";
+        internal const string ProdOtherEndpoint = @"https://api.xmpush.global.xiaomi.com";
+        internal const string SandboxEndpoint = @"https://sandbox.xmpush.xiaomi.com";
```
[the remark this answers is not in the document]
Comment (weihengSu2) [reply src/Microsoft.Azure.NotificationHubs/XiaomiCredential.cs:22]: for mockEndpoint, I guess we could wait for Louis to see if we have it? I could update it if we have or I could place a holder here

File: src/Microsoft.Azure.NotificationHubs/XiaomiCredential.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        public XiaomiCredential()
+            : base()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:Microsoft.Azure.NotificationHubs.XiaomiCredential"/> class.
+        /// </summary>
+        /// <param name="appSecret">The Xiaomi app secret key.</param>
+        public XiaomiCredential(string appSecret)
```
[the remark this answers is not in the document]
Comment (weihengSu2) [reply src/Microsoft.Azure.NotificationHubs/XiaomiCredential.cs:37]: Yes I could update it. Real quick

File: src/Microsoft.Azure.NotificationHubs/XiaomiCredential.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            {
+                return base.GetHashCode();
+            }
+
+            return unchecked(AppSecret.GetHashCode() ^ Endpoint.GetHashCode());
+        }
+
+        /// <summary>Validates the MPNS credential.</summary>
+        /// <param name="allowLocalMockPns">true to allow local mock PNS; otherwise, false.</param>
+        protected override void OnValidate(bool allowLocalMockPns)
```
[the remark this answers is not in the document]
Comment (weihengSu2) [reply src/Microsoft.Azure.NotificationHubs/XiaomiCredential.cs:122]: I could update it

File: src/Microsoft.Azure.NotificationHubs/NotificationHubDescription.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpodwysocki) [reply src/Microsoft.Azure.NotificationHubs/NotificationHubDescription.cs:336]: I don't think in this case the order matters.  See [Data Member Order](https://learn.microsoft.com/en-us/dotnet/framework/wcf/feature-details/data-member-order)

[the remark this answers is not in the document]
Comment (weihengSu2) [reply src/Microsoft.Azure.NotificationHubs/NotificationHubDescription.cs:336]: Ah Matt. Thanks very much for jumping into the review. Actually it should be 1015. I didn't pay attention to it before cuz 1007 has been used by another operation.

**Comment by weihengSu2** [comment]
I will address the comments in https://github.com/Azure/azure-notificationhubs-dotnet/pull/264. I have combined 260/261/262 in that PR

**Pull request merged**